src/carousel/carousel.ts

```diff
@@ -103,14 +105,38 @@ export class NgbCarousel implements AfterContentChecked,
   }
 
   private _cycleToNext() {
-    const nextIdx = this._activeIdx + 1;
-    this.select(nextIdx >= this._slides.length ? (this.wrap ? 0 : nextIdx - 1) : nextIdx);
+    let selectedId: string = this._getOtherSlide(this.activeId, true);
+    this.select(selectedId);
   }
 
   private _restartTimer() {
     this.pause();
     this.cycle();
   }
+
+  private _getSlideById(slideIdx: string): any {
+    let slideWithId: NgbSlide[] = this._slides.filter(slide => slide.id === slideIdx);
+    return slideWithId.length ? slideWithId[0] : null;
+  }
+
+  private _getOtherSlide(id: string, next: boolean): string {
```

Comment:
This methods looks quite complex. Could you think of writing it differently so it is easier to read / reason about? Maybe split it into 2 methods (`_getNextSlide` / `_getPrevSlide`)?